clicache/src/QueryService.hpp

```diff
@@ -51,7 +60,7 @@ namespace Apache
       /// <summary>
       /// Provides a query service.
       /// </summary>
-      generic<class TKey, class TResult>
+      generic<class TResult>
```

Comment:
The more I think about it does it even make sense for the class itself to be templated. Does the class itself depend in any way on the `TResult`. I think the answer is no if you look at each method. The bulk of the methods are factories for `Query<TResult>` and `CqQuery<TKey, TResult>`.  So should have to ask for a new `QueryService` every time I want to make new `Query` with a different `TResult`? That seems wrong to me.
```
var qsString = cache.getQueryService<string>();
var qString = qsString.createQuery("....");
var qsInt = cache.getQueryService<int>)();
var qInt = qsInt.createQuery("...");
```
vs.
```
var qs = cache.getQueryService();
var qString = qs.createQuery<string>("....");
var qInt = qs.createQuery<int>("...");
```

Reply:
Additionally methods like `CloseCqs` operates on all CQs, not just the ones that match the `TResult` of the class.

Reply:
I agree, so I will pull all of the templating out of QueryService.